main/solution/backend/serverless.yml

```diff
@@ -26,6 +26,9 @@ provider:
     restApi:
       format: '{"authorizer.principalId":"$context.authorizer.principalId","error.message":"$context.error.message","extendedRequestId":"$context.extendedRequestId","httpMethod":"$context.httpMethod","identity.sourceIp":"$context.identity.sourceIp","integration.error":"$context.integration.error","integration.integrationStatus":"$context.integration.integrationStatus","integration.latency":"$context.integration.latency","integration.requestId":"$context.integration.requestId","integration.status":"$context.integration.status","path":"$context.path","requestId":"$context.requestId","responseLatency":"$context.responseLatency","responseLength":"$context.responseLength","stage":"$context.stage","status":"$context.status"}'
       executionLogging: false
+  tracing:
+    apiGateway: true
+    lambda: true
```

Comment:
Does this ensure that serverless enables XRay at all APIGateway and Lambda resources? Are there additional permissions needed in IAM role that serverless uses to make this happen? I did notice that we are adding permissions to add events from Lambda or APIGateway, but this is more of a question on permissions required for original setup.

Reply:
Yep, this enables XRay for API GW and Lambda.
Are you referring to the IAM policy for CICD? If yes, then that role does have permissions to add XRay to Lambda and API GW.
https://github.com/awslabs/service-workbench-on-aws/blob/eb5b7eb7100c234a44e63720272e0c270314fce7/main/cicd/cicd-pipeline/config/infra/cloudformation.yml#L217

That role has the `PowerUserAccess` permission which gives it access to all AWS resources except management of IAM Users and Groups.

Reply:
Ah, that makes sense. We have a separate task of narrowing down scope of that role. Do you mind adding x-ray specific permissions here since we plan on removing PowerUserAccess. Consider this as a non-blocking comment and tackle it if you already know the specific permission and is something that can be quickly done.